### PR TITLE
Allow reserved spaces for local competitors in finals

### PR DESCRIPTION
**Pull request by ohexter**

### CODE REVIEW

File: wca-guidelines.md [hunk 1]
```diff
@@ -130,6 +130,8 @@ To be more informative, each Guideline is classified using one of the following
 - 9f4+) [CLARIFICATION] The result of an attempt should be a DNF if the competitor began the attempt by indicating that they are ready (see [Regulation A3b2](regulations:regulation:A3b2)) but the attempt was disqualified.
 - 9f5+) [CLARIFICATION] The result of an attempt is DNS if the competitor was eligible for the attempt and did not start it (see [Regulation A3b2](regulations:regulation:A3b2)). If the competitor did not qualify for an attempt (e.g. in a Cutoff Round), they have no result for the attempt.
 - 9p2+) [CLARIFICATION] Result refers to the single or average that determines the ranking of the corresponding round.
+- 9p2b+) [ADDITION] If there are not enough competitors representing the relevant region to fill the reserved spaces, they are replaced with the top ranked competitors representing any region.
```
Comment (ohexter) [inline wca-guidelines.md]:
```suggestion
- 9p2b+) [ADDITION] If there are not enough competitors representing the relevant region to fill the reserved spaces, they are replaced with the best competitors representing any region.
```

**Comment by Jobarion** [comment]
This doesn't seem to take into account rounds before the final round. If the final has reserved spaces, all previous rounds should also have reserved spaces with the same conditions as the final.

**Comment by ohexter** [comment]
This is certainly possible, but the problem statement has typically been expressed in terms of finals and I'm not aware of a need for further allowances.

**Comment by Jobarion** [comment]
Since qualifying for all previous rounds is a prerequisite for participating in a final I don't see a good reason not to reserve spaces in previous rounds as well. The goal of this change is to make sure at least N competitors from a given region will be in the final. This is not guranteed if nobody from that region qualifies for the semi final.

**Comment by CarterKoala** [comment]
I think it makes sense - it's just a minor wording change. It shouldn't matter for semifinals except in very extreme cases, but I think it still makes sense to allow for this.

File: wca-guidelines.md [hunk 2]
```diff
@@ -130,6 +130,8 @@ To be more informative, each Guideline is classified using one of the following
 - 9f4+) [CLARIFICATION] The result of an attempt should be a DNF if the competitor began the attempt by indicating that they are ready (see [Regulation A3b2](regulations:regulation:A3b2)) but the attempt was disqualified.
 - 9f5+) [CLARIFICATION] The result of an attempt is DNS if the competitor was eligible for the attempt and did not start it (see [Regulation A3b2](regulations:regulation:A3b2)). If the competitor did not qualify for an attempt (e.g. in a Cutoff Round), they have no result for the attempt.
 - 9p2+) [CLARIFICATION] Result refers to the single or average that determines the ranking of the corresponding round.
+- 9p2b+) [ADDITION] If there are not enough competitors representing the relevant region to fill the reserved spaces, they are replaced with the best competitors representing any region.
+- 9p2b++) [RECOMMENDATION] Any use of reserved spaces for final rounds should be clearly announced before the competition.
```
Comment (ohexter) [inline wca-guidelines.md]:
```suggestion
- 9p2b++) [RECOMMENDATION] Any use of reserved spaces should be clearly announced before the competition.
```

File: wca-regulations.md [hunk 3]
```diff
@@ -281,6 +281,7 @@ Note: Because Article and Regulation numbers are not reassigned when Regulations
     - 9p1) At least 25% of competitors must be eliminated between consecutive rounds of the same event.
     - 9p2) The competitors who advance to the next round must be determined by ranking (best X competitors) or by result (all competitors with a better result than X) in the preceding round.
         - 9p2a) For each round, advancement conditions must be announced before the round starts, and should not be changed after it has begun. Changes must be made at the discretion of the WCA Delegate, who must carefully consider the fairness of the change.
+        - 9p2b) Exception: For final rounds of a competition designated as a National Championship or Continental Championship, the organization team may reserve spaces for competitors representing the relevant region. A maximum of 8 spaces or 50% of spaces in the final round, whichever is the lower, may be reserved in this way. For a final round with reserved spaces, the competitors advancing from the previous round are a reserved number of best competitors representing the relevant region, followed by the remaining best competitors representing any region.
```
Comment (ohexter) [inline wca-regulations.md]:
```suggestion
        - 9p2b) Exception: For competitions designated as a National Championship or Continental Championship, the organization team may amend advancement conditions to reserve spaces for competitors representing the relevant region. A maximum of 8 spaces or 50% of spaces in the following round, whichever is the lower, may be reserved in this way. For a round with reserved spaces, the competitors advancing from the previous round are a reserved number of best competitors representing the relevant region, followed by the remaining best competitors representing any region.
```

**Review by ohexter — COMMENTED** [review]
Allow for all progression rather than just final rounds

File: wca-guidelines.md [hunk 4]
```diff
@@ -130,6 +130,9 @@ To be more informative, each Guideline is classified using one of the following
...
 - 9f5+) [CLARIFICATION] The result of an attempt is DNS if the competitor was eligible for the attempt and did not start it (see [Regulation A3b2](regulations:regulation:A3b2)). If the competitor did not qualify for an attempt (e.g. in a Cutoff Round), they have no result for the attempt.
 - 9p2+) [CLARIFICATION] Result refers to the single or average that determines the ranking of the corresponding round.
+- 9p2b+) [ADDITION] If there are not enough competitors representing the relevant region to fill the reserved spaces, they are replaced with the best competitors representing any region.
+- 9p2b++) [ADDITION] For a competition designated as both a World Championship and a Continental Championship and/or National Championship, this exception does not apply. For a competition designated as both a Continental Championship and a National Championship, places may only be reserved for competitors representing the relevant continent.
+- 9p2b+++) [RECOMMENDATION] Any use of reserved spaces for final rounds should be clearly announced before the competition.
```
Comment (ohexter) [inline wca-guidelines.md]:
```suggestion
- 9p2b+++) [RECOMMENDATION] Any use of reserved spaces should be clearly announced before the competition.
```

File: wca-guidelines.md [hunk 5]
```diff
@@ -130,6 +130,9 @@ To be more informative, each Guideline is classified using one of the following
 - 9f4+) [CLARIFICATION] The result of an attempt should be a DNF if the competitor began the attempt by indicating that they are ready (see [Regulation A3b2](regulations:regulation:A3b2)) but the attempt was disqualified.
 - 9f5+) [CLARIFICATION] The result of an attempt is DNS if the competitor was eligible for the attempt and did not start it (see [Regulation A3b2](regulations:regulation:A3b2)). If the competitor did not qualify for an attempt (e.g. in a Cutoff Round), they have no result for the attempt.
 - 9p2+) [CLARIFICATION] Result refers to the single or average that determines the ranking of the corresponding round.
+- 9p2b+) [ADDITION] If there are not enough competitors representing the relevant region to fill the reserved spaces, they are replaced with the best competitors representing any region.
+- 9p2b++) [ADDITION] For a competition designated as both a World Championship and a Continental Championship and/or National Championship, this exception does not apply. For a competition designated as both a Continental Championship and a National Championship, places may only be reserved for competitors representing the relevant continent.
```
Comment (ohexter) [inline wca-guidelines.md]:
```suggestion
- 9p2b++) [ADDITION] For a competition designated as both a World Championship and a Continental Championship and/or National Championship, this exception does not apply. For a competition designated as both a Continental Championship and a National Championship, spaces may only be reserved for competitors representing the relevant continent.
```

**Pull request merged**